pkg/cloud/rgraph/rnode/healthcheck/type_trait.go

```diff
@@ -41,7 +41,48 @@ func (*typeTrait) FieldTraits(meta.Version) *api.FieldTraits {
 	dt.OutputOnly(api.Path{}.Pointer().Field("Region"))
 	dt.OutputOnly(api.Path{}.Pointer().Field("SelfLink"))
 
-	// TODO: handle alpha/beta
+	// Optional fields
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Description"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("LogConfig"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("GrpcHealthCheck"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("GrpcHealthCheck").Pointer().Field("GrpcServiceName"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("GrpcHealthCheck").Pointer().Field("Port"))
+	// This field is not supported
+	dt.OutputOnly(api.Path{}.Pointer().Field("GrpcHealthCheck").Pointer().Field("PortName"))
+
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Http2HealthCheck"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Http2HealthCheck").Pointer().Field("Response"))
+	// This field is not supported
+	dt.OutputOnly(api.Path{}.Pointer().Field("Http2HealthCheck").Pointer().Field("PortName"))
+
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("HttpHealthCheck"))
+	// This field is not supported
+	dt.OutputOnly(api.Path{}.Pointer().Field("HttpHealthCheck").Pointer().Field("PortName"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("HttpHealthCheck").Pointer().Field("Response"))
+
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("HttpsHealthCheck"))
+	// This field is not supported
+	dt.OutputOnly(api.Path{}.Pointer().Field("HttpsHealthCheck").Pointer().Field("PortName"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("HttpsHealthCheck").Pointer().Field("Response"))
+
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("SslHealthCheck"))
+	// This field is not supported
+	dt.OutputOnly(api.Path{}.Pointer().Field("SslHealthCheck").Pointer().Field("PortName"))
```

Comment:
(in separate patch) should we consider a new type - `Unsupported`? Not sure if `Output` is the best choice here, even `System` looks better. API description also doesn't help - is it deprecated or a planned feature?

Reply:
This field is Deprecated but still this value might be already set and in this case it will be returned. OutputOnly indicates that this field might be returned by the cloud but should not be set by clients